### PR TITLE
Add description of schedule feature

### PR DESCRIPTION
**Pull request by Chesterwongz**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -154,6 +161,76 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+        
+    }
+    
+    // other SchedulePanel methods...
+}
+```
+When `SchedulePanel` is initialised, it sets up a `CalendarSource` and `DetailedWeekView` with our custom display settings.
+Any changes made to the `Calendar` in `Model` will automatically update the `DetailedWeekView` in the `SchedulePanel` through CalendarFX's internal implementation (see how in the [manual](http://dlsc.com/wp-content/html/calendarfx/manual.html)). 
+
+####Design considerations:
```
Comment (angkoonhwee) [inline docs/DeveloperGuide.md:220]: Somehow this isn't shown as a h4 header but like that `####Design`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -154,6 +161,76 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+**Aspect: How schedule is implemented:**
+
+* **Alternative 1:**  Create a calendar view using JavaFX.
+    * Pros: More customisable as we are not limited by CalendarFX's API.
+    * Cons: Much more difficult to implement.
+
+* **Alternative 2 (current choice):** Use CalendarFX to display entries while storing entry data locally.
```
Comment (angkoonhwee) [inline docs/DeveloperGuide.md:228]:
```suggestion
* **Alternative 2 (current implementation):** Use CalendarFX to display entries while storing entry data locally.
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -154,6 +161,76 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+        calendarView.getCalendarSources().addAll(calendarSource);
+        
+        calendarView.setStartTime(TimeRange.DAY_START);
+        calendarView.setEndTime(TimeRange.DAY_END);
+        // other settings...
+        
+    }
+    
+    // other SchedulePanel methods...
+}
```
Comment (angkoonhwee) [inline docs/DeveloperGuide.md:215]: Can give a simplified version of the code, e.g. remove the "adapted from" can also add comments to explain what the things mean `private final Logger logger = LogsCenter.getLogger(this.getClass());` can write a comment at the side to explain so the code snippet is clearer

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -154,6 +161,76 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
 
+### Schedule feature
+
+Users can view a calendar that contains all existing lessons to visualise their schedule and plan ahead.
+
+#### Current Implementation
+
+TAB uses the [CalendarFX](https://dlsc.com/products/calendarfx/) library to implement its schedule view.
+Each lesson stored in the `Model` component is also converted to a CalendarFX `Entry` and maintained in the `CalendarEntryList` of `AddressBook`. 
+The `Ui` component shows this list of entries in the `SchedulePanel` using CalendarFX's `DetailedWeekView`.
```
Comment (angkoonhwee) [inline docs/DeveloperGuide.md:172]: Can explain in more details like how `CalendarEntryList` was derived from lessons etc.

**Review by angkoonhwee — COMMENTED** [review]
I think can explain in more details how your caledarfx works >.<

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -23,80 +26,83 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
...
+* depends on some classes in the `Model` component, as it displays `Person` and `Calendar` objects residing in the `Model`.
+* chooses which component it displays based on the `CommandResult` returned by `Logic` component after executing the user command (e.g., `CommandResult#isShowSchedule()` tells the `Ui` to display the user's schedule. See also: [Schedule feature](#schedule-feature)). 
 
 ### Logic component
 
-**API** : [`Logic.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/logic/Logic.java)
+**API** : [`Logic.java`](https://github.com/AY2122S1-CS2103T-F13-3/tp/tree/master/src/main/java/seedu/address/logic/Logic.java)
 
-Here's a (partial) class diagram of the `Logic` component:
+Here's a (partial) *class diagram* of the `Logic` component:
```
Comment (lingshanng) [inline docs/DeveloperGuide.md:97]: is "class diagram" a glossary term?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -114,17 +120,18 @@ How the parsing works:
...
 <img src="images/ModelClassDiagram.png" width="800" />
 
 
-The `Model` component,
+The `Model` component
 
-* stores the address book data i.e., all `Person` objects (which are contained in a `UniquePersonList` object).
-* stores the currently 'selected' `Person` objects (e.g., results of a search query) as a separate _filtered_ list which is exposed to outsiders as an unmodifiable `ObservableList<Person>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
+* stores the tuition address book data, i.e., all `Person` and calendar `Entry` objects (which are contained in a `UniquePersonList` object and `CalendarEntryList` object respectively).
+* stores the currently 'selected' `Person` objects (e.g., results of a search query) as a separate _filtered_ list which is exposed to outsiders as an unmodifiable `ObservableList<Person>` that can be 'observed', e.g., the `Ui` component can be bound to this list so that it automatically updates when the data in the list change.
```
Comment (lingshanng) [inline docs/DeveloperGuide.md:132]: maybe remove italics from filtered

**Pull request merged**